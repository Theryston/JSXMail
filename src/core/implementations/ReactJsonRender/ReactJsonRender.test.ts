import React from 'react';
import { ReactJsonRender } from '.';

describe('ReactJsonRender', () => {
  it('should call fromComponent with correct arguments', () => {
    const reactJsonRender = new ReactJsonRender();
    const componentFunction = jest.fn(() => {
      return React.createElement('h1', null, 'Hello World!');
    });
    const props = {};
    reactJsonRender.fromComponent(componentFunction, props);
    expect(componentFunction).toHaveBeenCalledWith(props);
  });

  it('should return correct result', async () => {
    const reactJsonRender = new ReactJsonRender();
    const componentFunction = jest.fn(({ test }) => {
      function TestComponent({ text }) {
        return React.createElement('h1', null, text);
      }

      return React.createElement(
        'div',
        {
          width: '100px',
        },
        test,
        React.createElement(TestComponent, {
          text: 'Test Hello World!',
        }),
      );
    });

    const props = {
      test: 'testing',
    };

    const result = await reactJsonRender.fromComponent(
      componentFunction,
      props,
    );

    expect(result).toMatchSnapshot();
  });
});
