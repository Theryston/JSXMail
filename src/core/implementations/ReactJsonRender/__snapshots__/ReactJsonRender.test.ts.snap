// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactJsonRender should return correct result 1`] = `
Object {
  "children": Array [
    "testing",
    Object {
      "children": Array [
        "Test Hello World!",
      ],
      "props": Object {},
      "type": "h1",
    },
  ],
  "props": Object {
    "width": "100px",
  },
  "type": "div",
}
`;
